pkg/operator/ceph/cluster/predicate.go

```diff
@@ -21,19 +21,36 @@ import (
 	"context"
 
 	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
 	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
 	"github.com/rook/rook/pkg/clusterd"
 	discoverDaemon "github.com/rook/rook/pkg/daemon/discover"
 	"github.com/rook/rook/pkg/operator/ceph/controller"
 	"github.com/rook/rook/pkg/operator/k8sutil"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/event"
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 )
 
+func compareNodes(objOld, objNew *corev1.Node) string {
+	// do not watch node if only resourceversion got changed
+	resourceQtyComparer := cmpopts.IgnoreFields(v1.ObjectMeta{}, "ResourceVersion")
+	diff := cmp.Diff(objOld.Spec, objNew.Spec, resourceQtyComparer)
+
+	// do not watch node if only LastHeartbeatTime got changed
+	resourceQtyComparer = cmpopts.IgnoreFields(corev1.NodeCondition{}, "LastHeartbeatTime")
+	diff1 := cmp.Diff(objOld.Spec, objNew.Spec, resourceQtyComparer)
+
+	if diff == "" && diff1 == "" {
+		return diff
+	}
+	return "not-equal"
```

Comment:
How about returning a bool? The actual diff isn't being returned.

Reply:
Sure, updated